### PR TITLE
Support estimated count with multiple schemas

### PR DESCRIPTION
**Pull request by wolfogre**
The `EstimateCount` could be incorrect when the table lives in multiple schemas. Related to #19775.

See https://wiki.postgresql.org/wiki/Count_estimate:

> If you don't need an exact count, the current statistic from the catalog table pg_class might be good enough and is much faster to retrieve for big tables.
> `SELECT reltuples AS estimate FROM pg_class WHERE relname = 'table_name';`
> Tables named "table_name" can live in multiple schemas of a database, in which case you get multiple rows for this query. To overcome ambiguity:
> `SELECT reltuples::bigint AS estimate FROM pg_class WHERE oid = 'schema_name.table_name'::regclass;`

And it happened to me:

image

I don't think it's a bug, because it's correct in most cases. So I labeled this PR with `kind/enhancement`.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: models/db/context.go [hunk 1]
```diff
@@ -188,7 +188,9 @@ func EstimateCount(ctx context.Context, bean interface{}) (int64, error) {
 	case schemas.MYSQL:
 		_, err = e.Context(ctx).SQL("SELECT table_rows FROM information_schema.tables WHERE tables.table_name = ? AND tables.table_schema = ?;", tablename, x.Dialect().URI().DBName).Get(&rows)
 	case schemas.POSTGRES:
-		_, err = e.Context(ctx).SQL("SELECT reltuples AS estimate FROM pg_class WHERE relname = ?;", tablename).Get(&rows)
+		// the table can live in multiple schemas of a postgres database
```
Comment (delvh) [inline models/db/context.go:191]:
```suggestion
		// the table can live in multiple schemas of a postgres database
		// See https://wiki.postgresql.org/wiki/Count_estimate
```

File: models/db/context.go [hunk 2]
```diff
@@ -188,7 +188,9 @@ func EstimateCount(ctx context.Context, bean interface{}) (int64, error) {
...
+		tablename = x.TableName(bean, true)
+		_, err = e.Context(ctx).SQL("SELECT reltuples::bigint AS estimate FROM pg_class WHERE oid = ?::regclass;", tablename).Get(&rows)
```
Comment (delvh) [inline models/db/context.go:194]: Hmm, in the docs this is `schema_name.table_name::`, not `table_name::`.
Will this really work as intended?

Comment (wolfogre) [reply models/db/context.go:194]: The definition of `x.TableName(bean, true)`:

```go
// TableName returns table name with schema prefix if has
func (engine *Engine) TableName(bean interface{}, includeSchema ...bool) string {
	return dialects.FullTableName(engine.dialect, engine.GetTableMapper(), bean, includeSchema...)
}
```

And I have tested it, it works fine.

**Review by delvh — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Pull request merged**